File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.py
class Solution(object):
    def countGoodSubstrings(self, s):
        n=len(s)
        ans=0
        k=3
        l=0
        dici={}
        for r in range(n):
            if s[r] in dici:
                dici[s[r]]+=1
            else:
                dici[s[r]]=1
            if r-l==k:
                dici[s[l]]-=1
                if dici[s[l]]==0:
                    dici.pop(s[l])
                l+=1
            if len(dici)==k:
                ans+=1
        return ans

File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.java
class Solution {
    public static boolean fun(String s,int k) {
        int n = s.length();
        HashSet<Character> hs = new HashSet<>();
        for (int i = 0; i < n; i++) {
            hs.add(s.charAt(i));
        }
        if (hs.size() == k) {
            return true;
        } else {
            return false;
        }
    }
    public int countGoodSubstrings(String s) {
        int n = s.length();
        int sl = 3;
        int count = 0;
        for (int i = 0; i <= n - sl; i++) {
            int j = i + sl - 1;
            boolean temp = fun(s.substring(i, j + 1),3);
            if (temp) {
                count++;
            }

        }
        return count;        
    }
}